src/feature/control/control_getinfo.c

```diff
@@ -1436,6 +1437,37 @@ getinfo_helper_liveness(control_connection_t *control_conn,
   return 0;
 }
 
+/** Implementation helper for GETINFO: answers queries about shared random
+ * value. */
+STATIC int
+getinfo_helper_rephist(control_connection_t *control_conn,
+                       const char *question, char **answer,
+                       const char **errmsg)
+{
+  (void) control_conn;
+  (void) errmsg;
+  int result = -1;
+
+  if (!strcmp(question, "rephist/ntor/onion_handshakes_assigned")) {
+    result =
+      rep_hist_get_circuit_handshake_assigned(ONION_HANDSHAKE_TYPE_NTOR);
+  } else if (!strcmp(question, "rephist/ntor/onion_handshakes_requested")) {
+    result =
+      rep_hist_get_circuit_handshake_requested(ONION_HANDSHAKE_TYPE_NTOR);
+  } else if (!strcmp(question, "rephist/tap/onion_handshakes_assigned")) {
+    result =
+      rep_hist_get_circuit_handshake_assigned(ONION_HANDSHAKE_TYPE_TAP);
+  } else if (!strcmp(question, "rephist/tap/onion_handshakes_requested")) {
+    result =
+      rep_hist_get_circuit_handshake_requested(ONION_HANDSHAKE_TYPE_TAP);
+  }
+  /* Else statement here is unrecognized key so do nothing. */
```

Comment:
Shouldn't we send an error here? Spec specifies:

`  The server sends a 551 or 552 error on failure.`